common.go

```diff
@@ -0,0 +1,110 @@
+package cose
+
+import (
+	"errors"
+	"fmt"
+)
+
+// IntOrStr is a value that can be
+type IntOrStr struct {
```

Comment:
Is it necessary to export this struct? I don't see in this PR any API accepting a `IntOrStr`, but it's just used internally to encode/decode data.